### PR TITLE
Introduce expo-insights module for iOS

### PR DESCRIPTION
**Pull request by tsapeta**
# Why

We’re building out a set of features that offer developers insight into the usage data of their apps

# How

Created a new package with a module that sends an event on app launch.

# Test Plan

Confirmed the network requests are sent and their data is then available in BigQuery

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]
good first pass based on our discussed spec. next step is to add `client_timestamp_event_start`, `client_timestamp_event_end`, and `client_timestamp_event_log_request_sent` fields to the request and the android implementation

File: packages/expo-insights/ios/InsightsModule.swift [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  guard (200...299).contains(response.statusCode) else {
+    log.warn("Insights: Server responded with status code \(response.statusCode) for event \(eventName)")
+    return
+  }
+}
+
+/**
+ Gets the project ID from the manifest.
+ */
+private func getProjectId(manifest: [String: Any]) -> String? {
```
Comment (wschurman) [inline packages/expo-insights/ios/InsightsModule.swift:86]: this should use the `expo-manifests` package if possible.

File: packages/expo-insights/ios/InsightsModule.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        }
+      }
+    }
+  }
+
+  /**
+   Sends the `APP_LAUNCH` event.
+   */
+  private func dispatchLaunchEvent() async throws {
+    guard let manifest = appContext?.constants?.constants()["manifest"] as? [String: Any] else {
```
Comment (wschurman) [inline packages/expo-insights/ios/InsightsModule.swift:30]: this gets the build-time manifest which is in the classic format still (I think). Either way using the `expo-manifests` package should help here.

https://linear.app/expo/issue/ENG-8510/address-post-commit-comments-from-httpsgithubcomexpoexpopull21932

File: packages/expo-insights/ios/InsightsModule.swift (hunk 1 above)
Comment (tsapeta) [reply packages/expo-insights/ios/InsightsModule.swift:86]: Yeah, the plan was to just make a simple version at the beginning. Neither I nor Łukasz had time to do more before the conference 😅

Comment (wschurman) [reply packages/expo-insights/ios/InsightsModule.swift:86]: Ah, got it. Cool.

**Pull request merged**